File: ruby/bowling_game/bowling_game_test.rb
require 'minitest/autorun'
require_relative 'bowling_game'

class BowlingGameTest < Minitest::Test
  def setup
    @bg = BowlingGame.new
  end

  def test_all_zeros
    20.times { @bg.roll(0) }
    assert_equal 0, @bg.score
  end

  def test_all_one
    20.times { @bg.roll(1) }
    assert_equal 20, @bg.score
  end

  def test_one_spare
    @bg.roll(5)
    @bg.roll(5)
    @bg.roll(3)
    17.times { @bg.roll(0) }
    assert_equal 16, @bg.score
  end

  def test_one_strike
    @bg.roll(10)
    @bg.roll(3)
    @bg.roll(4)
    16.times { @bg.roll(0) }
    assert_equal 24, @bg.score
  end

  def test_parfect_game
    12.times { @bg.roll(10) }
    assert_equal 300, @bg.score
  end
end

File: php/bowling_game/bowling_game_test.php
<?php

require_once 'bowling_game.php';

use PHPUnit\Framework\TestCase;

class bowling_game_test extends TestCase {
    public function testPerfectGame() {
        $game = new BowlingGame();
        for ($i = 0; $i < 12; $i++) {
            $game->roll(10);
        }
        $this->assertSame(300, $game->score());
    }
}

?>
